File: 1. Basics Of Programming/1.3 isNumberAPrime/main.cpp
#include<iostream>
using namespace std;
bool isPrime(int n){
    for(int fac = 2; fac * fac <= n ; fac++){
        if(n % fac == 0){
            return false;
        }
    }
    return true;
}
int main(){
    int t;

    cin >> t;

    while(t-- > 0){
        int n ;
        cin >> n;

        cout << (isPrime(n) ? "prime\n" : "not prime\n");
    }
}

File: 1. Basics Of Programming/1.3 isNumberAPrime/Main.java
import java.util.Scanner;
public class Main{
    public static void main(String args[]){
        Scanner scn = new Scanner(System.in);

        int t = scn.nextInt();

        while(t-- != 0){
            int n = scn.nextInt();

            System.out.println(isPrime(n) ? "prime" : "not prime");
        }
    }
    public static boolean isPrime(int n){
        for(int fac = 2 ; fac * fac <= n ; fac++){
            if(n % fac == 0){
                return false;
            }
        }
        return true;
    }
}